autonomi/src/client/data_private.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2024 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use std::hash::{DefaultHasher, Hash, Hasher};
+
+use bytes::Bytes;
+use serde::{Deserialize, Serialize};
+use sn_evm::{Amount, EvmWallet};
+use sn_protocol::storage::Chunk;
+use tokio::task::JoinSet;
+
+use super::data::{GetError, PutError};
+use crate::client::{ClientEvent, UploadSummary};
+use crate::{self_encryption::encrypt, Client};
+
+/// Private data on the network can be accessed with this
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct PrivateDataAccess(Chunk);
+
+impl PrivateDataAccess {
+    pub fn to_hex(&self) -> String {
+        hex::encode(self.0.value())
+    }
+
+    pub fn from_hex(hex: &str) -> Result<Self, hex::FromHexError> {
+        let data = hex::decode(hex)?;
+        Ok(Self(Chunk::new(Bytes::from(data))))
+    }
+
+    /// Get a private address for [`PrivateDataAccess`]. Note that this is not a network address, it is only used for refering to private data client side.
+    pub fn address(&self) -> String {
+        hash_to_short_string(&self.to_hex())
+    }
```

Comment:
This is kind of an important design decision. Wanted to outline this here. Public data refers to a Network address which points to a DataMap. `PrivateDataAccess` is not an address it IS THE DATAMAP. As it it can be very big (up to max chunk size), we need a way to refer to it in the client side, thus using its hash.